nomic/embed.py

```diff
@@ -61,18 +76,104 @@ def text_api_request(texts: List[str], model: str, task_type: str, dimensionalit
         raise Exception((response.status_code, response.text))
 
 
-def text(texts: List[str], model: str = "nomic-embed-text-v1", task_type: str = "search_document", dimensionality: int = None, long_text_mode: str = "truncate"):
+@overload
+def text(
+    texts: list[str],
```

Comment:
nit: should we use `List` instead of `list`? IIRC python < 3.9 doesn't support typing using the `list` keyword

Reply:
I added `from __future__ import annotations` so these annotations are just strings and don't have to be valid at runtime. This module imports fine with python 3.8.

Reply:
ah TIL thank you!